presto-spi/src/main/java/com/facebook/presto/spi/type/Type.java

```diff
@@ -146,6 +146,11 @@
     boolean equalTo(Block leftBlock, int leftPosition, Block rightBlock, int rightPosition);
 
     /**
+     * Are the values in the specified blocks at the specified positions equal are not distinct?
```

Comment:
I think the documentation here needs to be a lot more precise: how does equalTo relate to nulls? how is it different from isNotDistinctFrom? how do either of these methods relate to the result of hash?